app/presenters/vacancies_presenter.rb

```diff
@@ -26,30 +26,59 @@ def any?
     decorated_collection.count.nonzero?
   end
 
-  def total_count_message(keyword)
-    if total_count == 1
-      return I18n.t('jobs.job_count_without_search', count: total_count) unless @searched
+  def search_heading(keyword: '', location: '')
+      case true
+      when (keyword && keyword.length > 0) && (location && location.length > 0)
+        total_count_message_with_keyword_location(keyword, location)
+      when keyword && keyword.length > 0
+        total_count_message_with_keyword(keyword)
+      when location && location.length > 0
+        total_count_message_with_location(location)
+      else
+        total_count_message()
+      end
+  end
 
-      I18n.t('jobs.job_count', keyword: keyword, count: number_with_delimiter(total_count))
-    else
-      return I18n.t('jobs.job_count_plural_without_search', count: total_count) unless @searched
+  def total_count_message
+    return I18n.t('jobs.job_count_without_search', count: total_count) if total_count == 1
 
-      I18n.t('jobs.job_count_plural', keyword: keyword, count: number_with_delimiter(total_count))
-    end
+    I18n.t('jobs.job_count_plural_without_search', count: number_with_delimiter(total_count))
   end
 
-  def total_count_message_with_location(location, keyword)
+  def total_count_message_with_keyword_location(keyword, location)
     return I18n.t(
-      'jobs.job_count_with_location_category',
-      count: number_with_delimiter(total_count),
-      location: location,
-      keyword: keyword
+      'jobs.job_count_with_location_keyword',
+      count: total_count,
```

Comment:
Just checking: deliberate or accidental getting rid of `number_with_delimiter`?

Reply:
yes that was deliberate as there is no point delimiting the number 1?

Reply:
although i made assumption that `number_with_delimiter` turn 1000 into 1,000, am i wrong?